content/post/kenobi-boot2-root-writeup.md

```diff
@@ -0,0 +1,268 @@
+OS:0%Q=)U1(R=Y%DF=N%T=40%IPL=164%UN=0%RIPL=G%RID=G%RIPCK=G%RUCK=G%RUD=G)IE(
+OS:R=Y%DFI=N%T=40%CD=S)
+
+Network Distance: 1 hop
+Service Info: Host: KENOBI; OSs: Unix, Linux; CPE: cpe:/o:linux:linux_kernel
+
+OS and Service detection performed. Please report any incorrect results at https://nmap.org/submit/ .
+Nmap done: 1 IP address (1 host up) scanned in 25.27 second
+```
+
+This scan immediately told me a ton about the host. It appeared to be an Ubuntu linux distro, running samba, proftpd and a webserver. This meant there were a ton of paths to begin investigating. Given the low barrier to entry I started by opening firefox to see what the website it was serving looked like.
+
+![Kenobi site index](/img/kenobi_http_index.png)
+
+I tried a few different URLs for fun and ended up getting lucky with `robots.txt` which pointed out that there was a static `admin.html` page.
+
+![Kenobi site admin](/img/kenobi_http_admin.png)
+
+Give that there were a few other targets, I decided to look into those before enumarating the webserver further. Samba seemed like a reasonable next choice and I began by enumarating both shares and users using two of the smb scripts provided with nmap.
+
+```
+root@kali:~/scans# nmap --script=smb-enum-shares.nse,smb-enum-users.nse -p 445 -oA smb $ATTACKTARGET 
+Starting Nmap 7.80 ( https://nmap.org ) at 2020-11-14 22:22 UTC
+Nmap scan report for ip-10-10-221-157.eu-west-1.compute.internal (10.10.221.157)
+Host is up (0.00018s latency).
+
+PORT    STATE SERVICE
+445/tcp open  microsoft-ds
+MAC Address: 02:98:C4:09:75:23 (Unknown)
+
+Host script results:
+| smb-enum-shares: 
+|   account_used: guest
+|   \\10.10.221.157\IPC$: 
+|     Type: STYPE_IPC_HIDDEN
+|     Comment: IPC Service (kenobi server (Samba, Ubuntu))
+|     Users: 1
+|     Max Users: <unlimited>
+|     Path: C:\tmp
+|     Anonymous access: READ/WRITE
+|     Current user access: READ/WRITE
+|   \\10.10.221.157\anonymous: 
+|     Type: STYPE_DISKTREE
+|     Comment: 
+|     Users: 0
+|     Max Users: <unlimited>
+|     Path: C:\home\kenobi\share
+|     Anonymous access: READ/WRITE
+|     Current user access: READ/WRITE
+|   \\10.10.221.157\print$: 
+|     Type: STYPE_DISKTREE
+|     Comment: Printer Drivers
+|     Users: 0
+|     Max Users: <unlimited>
+|     Path: C:\var\lib\samba\printers
+|     Anonymous access: <none>
+|_    Current user access: <none>
+|_smb-enum-users: ERROR: Script execution failed (use -d to debug)
+
+Nmap done: 1 IP address (1 host up) scanned in 0.58 second
+```
+
+This ended up exposing a potential `kenobi` user as well as identifying an anonyous share. I connected to this share and was happy to find a `logs.txt` file that ended up being a gold mine. leaking not only that `kenobi` was a user but a path to an ssh key. This also contained a, presumably up-to-date, copy of the smb config file that pointed to where the users home directory and the samba share was.
```

Comment:
This scan exposed a potential `kenobi` user and also identified an anonymous share. I connected to that share and was delighted to find a `logs.txt` file that turned out to be a gold mine, exposing the `kenobi` user and a path to an SSH key.

This also contained a possibly up-to-date copy of...